### PR TITLE
Introduce gcp translation(V3), translate document providers

### PR DESCRIPTION
**Pull request by olegkachur-e**
- Add TranslateDocumentOperator and TranslateDocumentBatchOperator operators.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: providers/src/airflow/providers/google/cloud/operators/translate.py [hunk 1]
```diff
@@ -978,3 +983,324 @@ def execute(self, context: Context):
         )
         hook.wait_for_operation_done(operation=operation, timeout=self.timeout)
         self.log.info("Model deletion complete!")
+
+
+class TranslateDocumentOperator(GoogleCloudBaseOperator):
+    """
+    Translate document provided.
```
Comment (VladaZakharova) [inline providers/src/airflow/providers/google/cloud/operators/translate.py:990]: Maybe we can rephrase it a little bit? Or add more details :)

File: providers/src/airflow/providers/google/cloud/operators/translate.py [hunk 2]
```diff
@@ -978,3 +983,324 @@ def execute(self, context: Context):
...
+        self.gcp_conn_id = gcp_conn_id
+        self.impersonation_chain = impersonation_chain
+
+    def execute(self, context: Context) -> dict:
+        hook = TranslateHook(
+            gcp_conn_id=self.gcp_conn_id,
+            impersonation_chain=self.impersonation_chain,
+        )
+        try:
+            self.log.info("Starting the document translation run")
```
Comment (VladaZakharova) [inline providers/src/airflow/providers/google/cloud/operators/translate.py:1115]:
```suggestion
            self.log.info("Starting the document translation")
```

File: providers/src/airflow/providers/google/cloud/operators/translate.py [hunk 3]
```diff
@@ -978,3 +983,324 @@ def execute(self, context: Context):
...
+                enable_rotation_correction=self.enable_rotation_correction,
+                location=self.location,
+                labels=self.labels,
+                model=self.model,
+                glossary_config=self.glossary_config,
+                timeout=self.timeout,
+                retry=self.retry,
+                metadata=self.metadate,
+            )
+            self.log.info("Document translation run complete")
```
Comment (VladaZakharova) [inline providers/src/airflow/providers/google/cloud/operators/translate.py:1133]:
```suggestion
            self.log.info("Document translation completed")
```

File: providers/src/airflow/providers/google/cloud/operators/translate.py [hunk 4]
```diff
@@ -978,3 +983,324 @@ def execute(self, context: Context):
...
+            format_conversions=self.format_conversions,
+            enable_shadow_removal_native_pdf=self.enable_shadow_removal_native_pdf,
+            enable_rotation_correction=self.enable_rotation_correction,
+            models=self.models,
+            glossaries=self.glossaries,
+            metadata=self.metadata,
+            timeout=self.timeout,
+            retry=self.retry,
+        )
+        self.log.info("Batch document translation job started.")
```
Comment (VladaZakharova) [inline providers/src/airflow/providers/google/cloud/operators/translate.py:1297]: Is there some exception that can be raised in this call?

File: providers/tests/system/google/cloud/translate/example_translate_document.py [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+    TranslateDocumentBatchOperator,
+    TranslateDocumentOperator,
+)
+
+DAG_ID = "gcp_translate_document"
+PROJECT_ID = os.environ.get("SYSTEM_TESTS_GCP_PROJECT", "default")
+ENV_ID = os.environ.get("SYSTEM_TESTS_ENV_ID", "default")
+REGION = "us-central1"
+RESOURCE_DATA_BUCKET = "airflow-system-tests-resources"
+DATA_OUTPUT_BUCKET_NAME = f"bucket_{DAG_ID}_{ENV_ID}".replace("_", "-").lower()
```
Comment (VladaZakharova) [inline providers/tests/system/google/cloud/translate/example_translate_document.py:45]: why do we need to add ".lower()"? I think all the names here will be in lower case by default, no?

**Review by MaksYermak — APPROVED** [review]

**Pull request closed without merging**